quickwit/quickwit-serve/src/elasticsearch_api/rest_handler.rs

```diff
@@ -852,8 +857,14 @@ async fn es_scroll(
     };
     let search_response: SearchResponse = search_service.scroll(scroll_request).await?;
     // TODO append_shard_doc depends on the initial request, but we don't have access to it
+
+    // Ideally, we would have wanted to reuse the setting from the initial search request.
+    // However, passing that parameter is cumbersome and not necessary:
+    // if we have a valid `scroll_id` it means that the search request was successful.
+    // We can therefore allow failed splits, it won't make any difference.
```

Comment:
i agree it's cumbersome, and i'm fine with it being left as TODO.
I'm not convinced by the argument that if we have a scroll id, search was successful. After a few pages, we need to do more search, which might hit a missing split (or fail for whatever reason), so an initial success only guarantees we're okay for so much calls.

Reply:
:( yes you are correct.